### PR TITLE
feat: Add (Cluster)WorkflowTemplate DSL

### PR DESCRIPTION
**Pull request by paguos**
- Add WorkflowTempate DSL
- Add ClusterWorkflowTempate DSL

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ pytest = "*"
 flexmock = "*"
 
 [packages]
-argo-workflows = ">=3.1.1"
+argo-workflows = ">=3.5"
```
Comment (binarycrayon) [inline Pipfile:13]: thanks! by the way I'll find time to update the sdk to argo 2.11.* soon

File: tests/workflow_templates/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*.yaml
```
Comment (binarycrayon) [inline tests/workflow_templates/.gitignore:1]: trying to wrap my head around this, is there a specific reason we needed to ignore yaml files here? 
workflow templates are generating yaml files?

**Review by binarycrayon — APPROVED** [review]
Overall looks good to me, just wondering about the reason we needed to ignore yaml files in workflow template dir.

As soon as that question is cleared I will merge this.

File: tests/workflow_templates/.gitignore (hunk 2 above)
Comment (paguos) [reply tests/workflow_templates/.gitignore:1]: Yes, if you look at the integration tests, they generate the template and then submit it as a regular workflow

**Pull request merged**